### PR TITLE
Control panel app

### PR DESCRIPTION
**Pull request by deadcyclo**
Adds control panel app demo

### CODE REVIEW

**Comment by gamerson** [comment]
hey @deadcyclo  we have a test now to make sure that all 4 build toolchains output the same number of files: https://travis-ci.org/rotty3000/blade/builds/116170369#L3315

so that is why this is failling.  Don't worry I will add the bndtools control panel app demo and resend this pull and that should fix the error.

**Comment by gamerson** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: gradle/blade.portlet.controlpanel/src/main/resources/WEB-INF/ignore [hunk 1]
```diff
@@ -0,0 +1 @@
+This file exists as a workaround for the FreeMarker template engine.
```
Comment (gamerson) [inline gradle/blade.portlet.controlpanel/src/main/resources/WEB-INF/ignore]: @rotty3000 is this related to recent ServletContextHelper changes?

File: gradle/blade.portlet.controlpanel/src/main/java/blade/portlet/ControlPanelAppPortlet.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+	public void destroy() {
+		PortletContext portletContext = getPortletContext();
+
+		ServletContextPool.remove(portletContext.getPortletContextName());
+
+		super.destroy();
+	}
+
+	@Override
+	public void init(PortletConfig portletConfig) throws PortletException {
```
Comment (gamerson) [inline gradle/blade.portlet.controlpanel/src/main/java/blade/portlet/ControlPanelAppPortlet.java]: The work done here, looks like it should be automated maybe?  @rotty3000

File: gradle/blade.portlet.controlpanel/src/main/resources/WEB-INF/ignore (hunk 1 above)
Comment (deadcyclo) [inline gradle/blade.portlet.controlpanel/src/main/resources/WEB-INF/ignore]: @gamerson this has been in the freemarker example as long as I can remember, so quite possibly this isn't even needed anymore? I just copied it to be on the safe side.

**Comment by deadcyclo** [comment]
Updated this with a new name to differenciate between controlpanel app and entry

**Comment by gamerson** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by gamerson** [comment]
@deadcyclo  this has been moved here: https://github.com/gamerson/liferay-blade-samples/pull/11
